glommio/src/lib.rs

```diff
@@ -197,51 +211,51 @@
 //!
 //! ## Current limitations
 //!
-//! Due to our immediate needs which are a lot narrower, we make
-//! the following design assumptions:
+//! Due to our immediate needs which are a lot narrower, we make the following
+//! design assumptions:
 //!
-//!  - NVMe. Supports for any other storage type is not even considered.
-//!    This allow us to use io uring's poll ring for reads and writes which
-//!    are interrupt free. This also assumes that one is running either `XFS`
-//!    or `Ext4` (an assumption that Seastar also makes)
+//!  - NVMe. Supports for any other storage type is not even considered. This
+//!    allow us to use io uring's poll ring for reads and writes which are
+//!    interrupt free. This also assumes that one is running either `XFS` or
+//!    `Ext4` (an assumption that Seastar also makes)
 //!
 //!  - A corollary to the above is that the CPUs are likely to be the
-//!    bottleneck, so this crate has a CPU scheduler but lacks an I/O
-//!    scheduler. That, however, would be a welcome addition.
+//!    bottleneck, so this crate has a CPU scheduler but lacks an I/O scheduler.
+//!    That, however, would be a welcome addition.
 //!
-//!  - A recent kernel is no impediment, so a fully functional I/O uring is
-//!    present. In fact, we require a kernel so recent that it doesn't event
-//!    exist: operations like `mkdir, ftruncate`, etc which are not present
-//!    in today's (5.8) `io_uring` are simply synchronous and we'll live
-//!    with the pain in the hopes that Linux will eventually add support for
-//!    them.
+//!  - A recent kernel is no impediment, as long as a fully functional I/O uring
+//!    is present. In fact, we require a kernel so recent that it doesn't event
+//!    exist: operations like `mkdir, ftruncate`, etc which are not present in
+//!    today's (5.8) `io_uring` are simply synchronous and we'll live with the
+//!    pain in the hopes that Linux will eventually add support for them.
 //!
 //! ## Missing features
 //!
 //! There are many. In particular:
 //!
-//! * There is no yet cross-shard communication nor ergonomic primitives to
+//! * There is not yet any cross-shard communication nor ergonomic primitives to
```

Comment:
This is actually false now. I have merged support for cross-shard communication.
We should remove this fully.